src/node/delay.rs

```diff
@@ -425,31 +422,46 @@ impl AudioProcessor for DelayReader {
             // find address of the frame in the ring buffer just before `position`
             let prev_position = position.floor();
             let (prev_block_index, prev_frame_index) =
-                self.find_frame_adress_at_position(prev_position);
+                DelayReader::find_frame_adress_at_position(prev_position, ring_size, ring_index);
 
             // find address of the frame in the ring buffer just after `position`
             let next_position = position.ceil();
             let (next_block_index, next_frame_index) =
-                self.find_frame_adress_at_position(next_position);
+                DelayReader::find_frame_adress_at_position(next_position, ring_size, ring_index);
 
             // as position is negative k will be what we expect
             let k = (position - position.floor()) as f32;
-            let k_inv = 1. - k;
-
-            // compute linear interpolation between prev and next for each channel
-            for channel_number in 0..number_of_channels {
-                let prev_sample =
-                    ring_buffer[prev_block_index].channel_data(channel_number)[prev_frame_index];
-                let next_sample =
-                    ring_buffer[next_block_index].channel_data(channel_number)[next_frame_index];
 
-                let value = k_inv * prev_sample + k * next_sample;
-
-                self.internal_buffer[channel_number] = value;
-            }
+            self.playback_infos[index] = (
+                prev_block_index,
+                prev_frame_index,
+                next_block_index,
+                next_frame_index,
+                k,
+            );
+        }
 
-            // populate output at index w/ internal_buffer
-            output.set_channels_values_at(index, &self.internal_buffer);
+        // render channels aligned
```

Comment:
Same for the AudioBufferSourceNode, I think this two-step way could be greatly improved (performance and lines of code-wise) by doing all in one step. But's let's improve gradually and merge this in the mean time

Reply:
Did it on the DelayNode, seems to improve pref a bit indeed